<ui version="4.0" >
 <author>Aaron Racicot</author>
 <class>AllVariablesNewGui</class>
 <widget class="QDialog" name="AllVariablesNewGui" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>516</width>
    <height>376</height>
   </rect>
  </property>
  <property name="minimumSize" >
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle" >
   <string>Variable editor</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" >
   <item>
    <widget class="QGroupBox" name="variableBox" >
     <property name="sizePolicy" >
      <sizepolicy vsizetype="Expanding" hsizetype="Preferred" >
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title" >
      <string>Variable definition</string>
     </property>
     <layout class="QVBoxLayout" >
      <item>
       <widget class="QWidget" native="1" name="nameWidget" >
        <layout class="QHBoxLayout" >
         <item>
          <widget class="QLabel" name="label" >
           <property name="minimumSize" >
            <size>
             <width>100</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize" >
            <size>
             <width>100</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="toolTip" >
            <string>A name for this variable. Use only alphanumeric characters</string>
           </property>
           <property name="text" >
            <string>Name:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit" />
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" native="1" name="sourceWidget" >
        <layout class="QHBoxLayout" >
         <item>
          <widget class="QLabel" name="label_4" >
           <property name="minimumSize" >
            <size>
             <width>100</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize" >
            <size>
             <width>100</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="toolTip" >
            <string>Where the variable definition derives from. 
An "expression" uses the Tekoa domain-specific language, 
"python class" is a path to a variable defined in a python module (e.g. urbansim.gridcell.population), 
and "primary attribute" refers to a primary attribute of the given dataset. </string>
           </property>
           <property name="text" >
            <string>Variable type:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox_2" >
           <item>
            <property name="text" >
             <string>expression</string>
            </property>
           </item>
           <item>
            <property name="text" >
             <string>primary attribute</string>
            </property>
           </item>
           <item>
            <property name="text" >
             <string>Python class</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="cbIndicatorUse" >
        <property name="toolTip" >
         <string>If checked, this variable will be available to compute over simulation runs as an indicator.</string>
        </property>
        <property name="text" >
         <string>Will this variable be used as an indicator?</string>
        </property>
        <property name="checked" >
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="cbModelUse" >
        <property name="toolTip" >
         <string>If checked, this variable will be available to include in model specifications.</string>
        </property>
        <property name="text" >
         <string>Will this variable be used in a model?</string>
        </property>
        <property name="checked" >
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QWidget" native="1" name="definitionWidget" >
        <layout class="QHBoxLayout" >
         <item>
          <widget class="QLabel" name="label_5" >
           <property name="sizePolicy" >
            <sizepolicy vsizetype="Fixed" hsizetype="Preferred" >
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize" >
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize" >
            <size>
             <width>16777215</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="text" >
            <string>Definition:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QTextEdit" name="textEdit" >
           <property name="sizePolicy" >
            <sizepolicy vsizetype="Expanding" hsizetype="Expanding" >
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="baseSize" >
            <size>
             <width>0</width>
             <height>20</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" native="1" name="widget" >
     <property name="maximumSize" >
      <size>
       <width>16777215</width>
       <height>45</height>
      </size>
     </property>
     <layout class="QHBoxLayout" >
      <item>
       <widget class="QPushButton" name="cboCheckSyntax" >
        <property name="toolTip" >
         <string>Verify that the syntax of variable definition parses correctly.</string>
        </property>
        <property name="text" >
         <string>Check syntax</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="cboCheckData" >
        <property name="toolTip" >
         <string>Verify that the variable can be computed against the baseyear data for this project.</string>
        </property>
        <property name="text" >
         <string>Check against data</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer>
        <property name="orientation" >
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0" >
         <size>
          <width>441</width>
          <height>24</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="Line" name="line" >
        <property name="orientation" >
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="saveChanges" >
        <property name="enabled" >
         <bool>true</bool>
        </property>
        <property name="toolTip" >
         <string>Accept the new variable and return to the Variable Library.</string>
        </property>
        <property name="text" >
         <string>Accept</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="cancelWindow" >
        <property name="toolTip" >
         <string>Return to the Variable Library without saving.</string>
        </property>
        <property name="text" >
         <string>Cancel</string>
        </property>
       </widget>
      </item>
     </layout>
     <zorder>cboCheckSyntax</zorder>
     <zorder>line</zorder>
     <zorder>saveChanges</zorder>
     <zorder>cancelWindow</zorder>
     <zorder>cboCheckData</zorder>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
